src/Shared/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs

```diff
@@ -35,6 +37,15 @@ public class AspNetRequestIpLayoutRenderer : AspNetLayoutRendererBase
         /// <docgen category='Rendering Options' order='10' />
         public bool CheckForwardedForHeader { get; set; }
 
+        /// <summary>
+        /// Gets or sets the array index of the X-Forwarded-For header to use, if the desired client IP is not at
+        /// the zeroth index.  Defaults to zero.  If the index is too large the last array element is returned instead.
+        /// If a negative index is used, this is used as the position from the end of the array.
+        /// Minus one will indicate the last element in the array.  If the negative index is too large the first index
+        /// of the array is returned instead.
+        /// </summary>
+        public int Index { get; set; } = 0;
```

Comment:
With all the nice contributions that you have made, then I'm very eager to help you out.

But I think `Index` is a little too generic for such an exotic option.

Could it be called `CheckForwardedForHeaderOffset` ? And be implemented like this (where activates offset +  header-lookup at the same time):

```c#
public int CheckForwardedForHeaderOffset
{
     get => _checkForwardedForHeaderOffset;
     set
     {
           _checkForwardedForHeaderOffset = value;
           CheckForwardedForHeader = true;
     }
}
```
And can be used like this:
```xml
${aspnet-request-ip:CheckForwardedForHeaderOffset=1}
```

Reply:
great idea, i will do that.  And thank you very much for all the help you provide me.

Reply:
Completed